src/server/scripts/Commands/cs_anticheat.cpp

```diff
@@ -85,11 +97,56 @@ class anticheat_commandscript : public CommandScript
         Player* pTarget = player->GetConnectedPlayer();
 
         // teleport both to jail.
-        WorldLocation jail {1, 16226.5f, 16403.6f, -64.5f, 3.2f};
-        pTarget->TeleportTo(jail);
-        handler->GetPlayer()->TeleportTo(jail);
-        pTarget->SetHomebind(jail, 876);
+        handler->GetSession()->GetPlayer()->TeleportTo(1, 16226.5f, 16403.6f, -64.5f, 3.2f);
+
+        WorldLocation loc;
+        loc = WorldLocation(1, 16226.5f, 16403.6f, -64.5f, 3.2f);// GM Jail Location
+        pTarget->TeleportTo(loc);
+        pTarget->SetHomebind(loc, 876);// GM Jail Homebind location
+        pTarget->CastSpell(pTarget, SHACKLES);// shackle him in place to ensure no exploit happens for jail break attempt
+        Aura* dungdesert = pTarget->AddAura(LFG_SPELL_DUNGEON_DESERTER, pTarget);// LFG_SPELL_DUNGEON_DESERTER
```

Comment:
AddAura() can return null

Reply:
i believe i got it fully checked in https://github.com/TrinityCore/TrinityCoreCustomChanges/pull/73/commits/278938b5b14ff8b29ecdfc6def14b138fe02973a now